File: Python/MODULO/Divide_Conquer_Pow.py
def power(a, m, n):
    b = 1
    a %= n
    while m > 0:
        if m & 1:
            b = (b * a) % n
        m >>= 1
        a = (a * a) % n
    return b

a, m, n = 419, 6217, 6217
print("b = {}^{} mod {} = {}".format(a, m, n, power(a, m, n)))

File: Cpp/MODULO/Divide_Conquer_Pow.cpp
#include <bits/stdc++.h>
using namespace std;
#define int long long

int power(int a, int m, int n) {
    int b = 1;
    a %= n;
    while (m > 0) {
        if (m & 1) {
            b = (b * a) % n;
        }
        m >>= 1;
        a = (a * a) % n;
    }
    return b;
}

main() {
    int a = 419, m = 6217, n = 6217;
    cout << "b = " <<  a << "^" << m << " mod " << n << " = " << power(a, m, n);
    return 0;
}